### ISSUE TITLE
Add container exit error code between ADD and ContainerDied

### ISSUE THREAD

**Issue by karan** [kubernetes/kubernetes#95932]
**What would you like to be added**:

Between

```
SyncLoop (ADD, "api")
```

and

```
SyncLoop (PLEG): "foo(96eb6e9d-84cf-12ea-b64a-42010a80005d)", event: &pleg.PodLifecycleEvent{ID:"96eb6e9d-84cf-12ea-b64a-42010a80005d", Type:"ContainerStarted", Data:"bar"}
```

and

```
SyncLoop (PLEG): "foo(96eb6e9d-84cf-12ea-b64a-42010a80005d)", event: &pleg.PodLifecycleEvent{ID:"96eb6e9d-84cf-12ea-b64a-42010a80005d", Type:"ContainerDied", Data:"bar"}
```

We can't tell if container died with a status error (we can by looking at containerd logs though).

**Why is this needed**:

It would make debugging dying containers much easier.

**Comment by adisky** [comment]
@karan I am new to kubelet code, trying to understand.
 IIUC, PLEG does not passes `exitCode` in the generated events, one way is to pass the exit code in events generated.
https://github.com/kubernetes/kubernetes/blob/bce5d7a0158444080ed95aae6a5684ce4dd9444b/pkg/kubelet/pleg/generic.go#L223-L224

other way is to get the exit code from podCache before logging which return pod status.
https://github.com/kubernetes/kubernetes/blob/bce5d7a0158444080ed95aae6a5684ce4dd9444b/pkg/kubelet/kubelet.go#L1958-L1961

**Comment by adisky** [comment]
closing this issue, as It has been fixed by #99880
/close
